### PR TITLE
[ARCTIC-1288][ams] limit the total files size bytes for Optimizing

### LINKED ISSUE #1288: [Improvement]: `Full-Optimizing` occupies all optimizer resources making other tables' `Self-Optimizing` hang up
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Full-Optimizing takes a long time to optimize and occupies all resources, for example, it takes 1h 17min to rewrite ~40G files, making other tables' Self-Optimizing hang up

image

### How should we improve?

1. We should limit the resource occupation for each `Self-Optimizing`, like limiting the total file size for each plan, a moderate default value may be 5~10G 
2. The big file( file size >= 0.95 target size?) not related to any delete-files should not be rewritten when executing Full-Optimizing

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

fix #1288 

## Brief change log

  - add table property `self-optimizing.max-file-size-bytes` with default value 8 GB
  - optimizing plan with max file size bytes limit, and skipping some partitions
  - Mixed format table's Minor-Optimizing only calculates the file size of the Change Store
  - Iceberg format table's Minor-Optimizing only calculates the file size of small files/pos-delete files/equ-delete files

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/AbstractOptimizePlan.java [hunk 1]
```diff
@@ -113,16 +114,25 @@ protected List<BasicOptimizeTask> collectTasks(List<String> partitions) {
...
+
+  protected long getTaskFileSize(BasicOptimizeTask task) {
+    return task.getBaseFileSize() + task.getInsertFileSize() + task.getDeleteFileSize() + task.getPosDeleteFileSize();
+  }
+
+  protected int getTaskFileCount(BasicOptimizeTask task) {
+    return task.getBaseFileCnt() + task.getDeleteFileCnt() + task.getInsertFileCnt() + task.getPosDeleteFileCnt();
   }
 
   private boolean reachMaxFileCount() {
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/AbstractOptimizePlan.java:134]: I recommend changing the method name to `reachOptimizeLimit`.

Comment (wangtaohz) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/AbstractOptimizePlan.java]: OK, I changed it.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Comment by wangtaohz** [comment]
cherry-picked to 0.4.x

**Pull request merged**